juju/juju
# Merge 2.9

Merge 2.9

https://github.com/juju/juju/pull/14662
https://github.com/juju/juju/pull/14626
https://github.com/juju/juju/pull/14658
https://github.com/juju/juju/pull/14660
https://github.com/juju/juju/pull/14659
https://github.com/juju/juju/pull/14667
https://github.com/juju/juju/pull/14669
https://github.com/juju/juju/pull/14668
https://github.com/juju/juju/pull/14674
https://github.com/juju/juju/pull/14663
https://github.com/juju/juju/pull/14679
https://github.com/juju/juju/pull/14680
https://github.com/juju/juju/pull/14684
https://github.com/juju/juju/pull/14610
https://github.com/juju/juju/pull/14681
https://github.com/juju/juju/pull/14675
https://github.com/juju/juju/pull/14688
https://github.com/juju/juju/pull/14635
https://github.com/juju/juju/pull/14670
https://github.com/juju/juju/pull/14691
https://github.com/juju/juju/pull/14686
https://github.com/juju/juju/pull/14682
https://github.com/juju/juju/pull/14689
https://github.com/juju/juju/pull/14690
https://github.com/juju/juju/pull/14695

Conflicts
```
#       .github/workflows/smoke.yml
#       .github/workflows/snap.yml
#       Makefile
#       apiserver/facades/agent/provisioner/provisioninginfo.go
#       apiserver/facades/agent/provisioner/provisioninginfo_test.go
#       apiserver/facades/client/application/application.go
#       apiserver/facades/client/application/application_test.go
#       apiserver/facades/client/application/get_test.go
#       apiserver/facades/client/charms/client.go
#       apiserver/facades/client/charms/clientnormalize_test.go
#       apiserver/facades/client/charms/conversions.go
#       apiserver/facades/client/charms/interfaces/downloader.go
#       apiserver/facades/client/client/client_test.go
#       apiserver/facades/client/resources/facade.go
#       apiserver/facades/schema.json
#       caas/kubernetes/provider/application/application.go
#       caas/kubernetes/provider/constants/constants.go
#       cloudconfig/providerinit/providerinit_test.go
#       cloudconfig/userdatacfg.go
#       cloudconfig/userdatacfg_test.go
#       cloudconfig/userdatacfg_unix.go
#       cmd/containeragent/unit/agent.go
#       cmd/juju/application/deploy_test.go
#       cmd/juju/application/deployer/charm.go
#       cmd/juju/application/deployer/series_selector.go
#       cmd/juju/application/refresh.go
#       cmd/juju/application/refresher/refresher.go
#       cmd/juju/application/refresher/refresher_test.go
#       cmd/juju/application/utils/origin.go
#       cmd/juju/charmhub/convert_test.go
#       cmd/juju/charmhub/info_test.go
#       cmd/juju/commands/upgrademodel_test.go
#       cmd/juju/machine/list_test.go
#       cmd/juju/machine/show_test.go
#       cmd/juju/status/status_internal_test.go
#       cmd/plugins/juju-metadata/addimage_test.go
#       cmd/plugins/juju-metadata/listimages_test.go
#       core/bundle/changes/changes_test.go
#       core/charm/repository/charmhub.go
#       core/series/supportedseries.go
#       environs/bootstrap/prepare_test.go
#       environs/config/config.go
#       go.mod
#       go.sum
#       provider/ec2/local_test.go
#       rpc/params/image_metadata.go
#       rpc/params/internal.go
#       rpc/params/params_test.go
#       state/allwatcher_internal_test.go
#       testing/factory/factory_test.go
#       upgrades/operations.go
#       upgrades/upgrade_test.go
#       upgrades/upgradevalidation/version_test.go
#       worker/provisioner/provisioner_task_test.go
```

## QA steps

See PRs.

Smoke test, deploy the following, run status
```
applications:
  juju-qa-test:
    charm: juju-qa-test
    channel: stable
    num_units: 2
  nrpe:
    charm: cs:nrpe-64
    channel: stable
  ntp:
    charm: cs:ntp-40
    channel: stable
    series: focal
  telegraf:
    charm: cs:telegraf-37
    channel: stable
  ubuntu:
    charm: ubuntu
    series: focal
    num_units: 2
      #    to:
      #      - "lxd:new"
relations:
- - ntp:juju-info
  - juju-qa-test:juju-info
- - ntp:juju-info
  - ubuntu:juju-info
- - nrpe:general-info
  - ubuntu:juju-info
- - telegraf:juju-info
  - ubuntu:juju-info
```
